rskj-core/src/main/java/co/rsk/rpc/ExecutionBlockRetriever.java

```diff
@@ -71,6 +72,27 @@ public Block getExecutionBlock(String bnOrId) {
             return cachedBlock;
         }
 
-        throw new JsonRpcUnimplementedMethodException("Method only supports 'latest' and 'pending' as parameters so far.");
+        // Is the block specifier either a hexadecimal or decimal number?
+        Optional<Long> executionBlockNumber = Optional.empty();
+
+        if (Utils.isHexadecimalString(bnOrId)) {
```

Comment:
This looks good, but I would prefer using [`Long.decode()`](https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#decode(java.lang.String)). We're reimplementing two of the three `decode`'s decoding formats!

Reply:
Sorry about not responding this before. True, `Long.decode` already allows for parsing decimal, hexadecimal and octal, and I think that's cool. Nevertheless, I think we should enforce strict control on the functions we use to parse data that comes directly from a user request. In this case, if for some reason we overlook changes to `Long.decode` at some point, we could end up permitting input that we wouldn't originally want to allow. Does that make sense at all?